x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/RestEsqlStopAsyncAction.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.action;
+
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.Scope;
+import org.elasticsearch.rest.ServerlessScope;
+import org.elasticsearch.rest.action.RestRefCountedChunkedToXContentListener;
+import org.elasticsearch.xpack.core.async.AsyncStopRequest;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+import static org.elasticsearch.xpack.esql.action.EsqlQueryResponse.DROP_NULL_COLUMNS_OPTION;
+import static org.elasticsearch.xpack.esql.formatter.TextFormat.URL_PARAM_DELIMITER;
+
+@ServerlessScope(Scope.PUBLIC)
+public class RestEsqlStopAsyncAction extends BaseRestHandler {
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(POST, "/_query/async/{id}/stop"));
```

Comment:
Should it instead be `/_query/async/stop/{id}` ?
The ID is quite long and it can be hard to see the "verb". 

On the other hand, having the "stop" at the end would make it nicer if we ever need to add a query param to this endpoint. For example will we ever need to add something like "wait_for_completion_timeout=10s" or something? 

Or since this is a POST, maybe that would be in the body of the request so it's easier to see regardless?

Reply:
If you follow REST guidelines, I think it should be `<id>/stop` since you first identify the entity and then you say what you want to do with it. However, I admit there's usability aspect to it maybe, so maybe PM should have a say.

> For example will we ever need to add something like "wait_for_completion_timeout=10s" or something?

It is possible, though unclear - that's to be decided for phase 2.